src/monitor/monitor.c

```diff
@@ -1510,7 +1470,7 @@ static void signal_offline_reset(struct tevent_context *ev,
             service_signal_sysbus_reconnect(cur_svc);
         }
     }
-    signal_res_init(monitor);
+    signal_res_init();
```

Comment:
If I read the code correctly, the only possible execution chain is: `signal_offline_reset()` -> `signal_res_init()` -> `res_init()`
And `signal_offline_reset()` has a comment:
```
    /* Handle SIGUSR2 (tell all providers to go reset offline) */
    ...
    tes = tevent_add_signal(ctx->ev, ctx, SIGUSR2, 0,
                            signal_offline_reset, ctx);
```
This entire chain doesn't work anymore (btw, I'm not sure it was working as expected - did `service_signal_dns_reload()` really force `sssd_be` to go online unconditionally or did it merely forced a res_init()?).

tl,dr: handling of `SIGUSR2` should be either moved to `sssd_be` as well (and documentation - `man sssd` - corrected) or dropped. I don't know if this feature is really usefull (and if it is documented downstream).

@sumit-bose, @pbrezina, what would you say?

Reply:
AFAIK SIGUSR2 causes two things in providers:
- it calls data_provider_reset_offline() - try to go online immediately
- and data_provider_res_init() - reread resolv.conf and try to go online immediately

It is something that works, since I am using it from time to time. Its probably not useful for production, but definitely useful for testing. It should be preserved.

Reply:
Note that res_init is called after check_online, so the order should probably switch. But we recreate cares channel before check_online and we use cares for name resolution in providers so its probably not a big deal.

Reply:
Hi,

so far `SIGUSR1` and `SIGUSR2` worked well for me and is documented in the man page. So imo the functionality should not be dropped for the time being.

With the plan of dropping the monitor it might be worth to add options to `sssctl` and use SBus directly instead of signals?

bye,
Sumit

Reply:
> We can still do [c52a5a6](https://github.com/SSSD/sssd/commit/c52a5a640f0574f28281dd62238ffc7303eb4391) in 2.8.2 so at least the issue is fixed.

I think this is a good idea.

Reply:
> We can still do [c52a5a6](https://github.com/SSSD/sssd/commit/c52a5a640f0574f28281dd62238ffc7303eb4391) in 2.8.2 so at least the issue is fixed.

@aplopez, could you open a PR please?

Reply:
#6475

Reply:
This is the new [PR #6475](https://github.com/SSSD/sssd/pull/6475).

Reply:
Coming back to  `SIGUSR2`.

On one side, the backends already handle this signal. In that case they invoke `signal_be_reset_offline()` which is a wrapper around `check_if_online()`.

On the other side, the backends also provide a D-Bus method called `resetOffline()` which is invoked by 'signal_offline_reset()` from the monitor when it receives a `SIGUSR2` signal. So my first guess was that this is still working.

I just ran a test and I can now confirm that sending `SIGUSR1` and `SIGUSR2` to the monitor does work on the backends.

Reply:
Ok, ` SIGUSR2` hould work indeed.

But `signal_res_init()` -> `res_init()` leftover (and corresponding include) isn't needed anymore.

Reply:
Right. I left it thinking it was still needed by monitor, but I was wrong.